core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -615,6 +622,44 @@ class SnappyStoreHiveCatalog(context: SnappyContext)
     }
   }
 
+  def alterTableToAddIndexProp(inTable: QualifiedTableName, indexName: String): Unit = {
+    alterTableLock.synchronized {
+      val hiveTable = inTable.getTable(client)
+      var indexes = ""
+      try {
+        indexes = hiveTable.serdeProperties(ExternalStoreUtils.INDEX_NAME) + ","
+      } catch {
+        case e: NoSuchElementException =>
+      }
+      client.alterTable(
+        hiveTable.copy(serdeProperties = hiveTable.serdeProperties +
+          (ExternalStoreUtils.INDEX_NAME -> (indexes + indexName)))
+      )
+    }
+    cachedDataSourceTables.invalidate(inTable)
+  }
+
+  def alterTableToRemoveIndexProp(inTable: QualifiedTableName, indexName: String): Unit = {
+    alterTableLock.synchronized {
+      val hiveTable = inTable.getTable(client)
```

Comment:
see the comment before about using QualifiedTableName for indexName rather than String.